cumulusci/core/config/OrgConfig.py

```diff
@@ -314,6 +318,62 @@ def get_orginfo_cache_dir(self, cachename):
         cache_dir.mkdir(parents=True, exist_ok=True)
         return open_fs_resource(cache_dir)
 
+    def get_access_token(self, **userfields):
+        """Get the access token for a specific user
+
+        If no keyword arguments are passed in, this will return the
+        access token for the default user. If userfields has the key
+        "username", the access token for that user will be returned.
+        Otherwise, a SOQL query will be made based off of the
+        passed-in fields to find the username, and the token for that
+        username will be returned.
+
+        Values are cached; the cache can be reset with the 'Refresh
+        oauth token' keyword.
+
+        Examples:
+
+        | # default user access token:
+        | token = org.get_access_token()
+
+        | # access token for 'test@example.com'
+        | token = org.get_access_token(username='test@example.com')
+
+        | # access token for user based on lookup fields
+        | token = org.get_access_token(alias='dadvisor')
+
+        """
+        if not userfields:
+            # No lookup fields specified? Return the token for the
+            # default user
+            return self.access_token
+
+        cache_key = str(sorted(userfields.items()))
+        if cache_key not in self._access_token_cache:
+            # if we have a username, use it. Otherwise we need to do a
+            # lookup using the passed-in fields.
+            username = userfields.get("username", None)
+            if username is None:
+                where = [f"{key} = '{value}'" for key, value in userfields.items()]
+                query = f"SELECT Username FROM User WHERE {' AND '.join(where)}"
+                result = self.salesforce_client.query_all(query).get("records", [])
+                if len(result) == 0:
+                    raise Exception("Couldn't find a username for the specified user.")
+                elif len(result) > 1:
+                    raise Exception("More than one user matched the search critiera.")
+                else:
+                    username = result[0]["Username"]
+
+            p = sfdx(f"force:org:display --targetusername={username} --json")
+            info = json.loads(p.stdout_text.read())
```

Comment:
This should probably go below the returncode check. If the command was unsuccessful it's more likely that the output is not valid json